src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -97,36 +97,85 @@ double get_temporal_id_value(const T& id) {
 CREATE_IS_CALLABLE_V(apply)
 
-// apply_callback accomplishes the overload for the
-// two signatures of callback functions.
+template <typename Callbacks, typename Box, typename Cache, typename TemporalId>
+struct AllSimpleCallbacks;
+
+template <typename... Callbacks, typename Box, typename Cache,
+          typename TemporalId>
+struct AllSimpleCallbacks<tmpl::list<Callbacks...>, Box, Cache, TemporalId>
+    : public std::integral_constant<
+          bool, tmpl2::flat_all_v<is_apply_callable_v<Callbacks, Box, Cache,
+                                                      TemporalId>...>> {};
+
 template <typename T, typename DbTags, typename Metavariables,
           typename TemporalId>
-bool apply_callback(
+bool apply_callbacks(
     const gsl::not_null<db::DataBox<DbTags>*> box,
     const gsl::not_null<Parallel::GlobalCache<Metavariables>*> cache,
     const TemporalId& temporal_id) {
-  if constexpr (is_apply_callable_v<typename T::post_interpolation_callback,
-                                    decltype(*box), decltype(*cache),
-                                    decltype(temporal_id)>) {
-    T::post_interpolation_callback::apply(*box, *cache, temporal_id);
+  using callbacks = typename T::post_interpolation_callbacks;
+  constexpr bool all_simple_callbacks =
+      AllSimpleCallbacks<callbacks, decltype(*box), decltype(*cache),
+                         decltype(temporal_id)>::value;
+  if constexpr (all_simple_callbacks) {
+    tmpl::for_each<callbacks>([&](const auto callback_v) {
+      using callback = tmpl::type_from<decltype(callback_v)>;
+      callback::apply(*box, *cache, temporal_id);
+    });
 
     // For the simpler callback function, we will always clean up volume data,
     // so we return true here.
     return true;
   } else {
-    return T::post_interpolation_callback::apply(box, cache, temporal_id);
+    static_assert(tmpl::size<callbacks>::value == 1,
+                  "Can only have 1 post_interpolation_callbacks that mutates "
+                  "the target box.");
+    return tmpl::front<callbacks>::apply(box, cache, temporal_id);
   }
 }
 
 CREATE_HAS_STATIC_MEMBER_VARIABLE(fill_invalid_points_with)
 CREATE_HAS_STATIC_MEMBER_VARIABLE_V(fill_invalid_points_with)
 
+template <typename T, bool B>
+struct with_value;
+
+template <typename T>
+struct with_value<T, false> {
+  using type = NoSuchType;
+};
+
+template <typename T>
+struct DoubleWrapper {
+  static constexpr double value = T::fill_invalid_points_with;
+};
+
+template <typename T>
+struct with_value<T, true> {
+  using type = DoubleWrapper<T>;
+};
+
+template <typename Callback>
+struct get_value {
+  using type =
+      typename with_value<Callback,
+                          has_fill_invalid_points_with_v<Callback>>::type;
+};
+
+template <typename Callbacks>
+struct get_invalid_fill {
+  using type =
+      tmpl::find<tmpl::flatten<tmpl::transform<Callbacks, get_value<tmpl::_1>>>,
+                 tt::is_a<DoubleWrapper, tmpl::_1>>;
```

Comment:
Is it worth checking for conflicting fill requests?  The code to do that might be kind of a pain.

Reply:
No I don't think it's worth it because all of the current callbacks that specify this fill value set it to `quiet_NaN`. And yeah it was going to be a pain to make it general so I just didn't.